core/packages/contracts/src/Gateway.sol

```diff
@@ -416,7 +420,14 @@ contract Gateway is IGateway {
         address assetHubAgent = $.agents[ASSET_HUB_AGENT_ID];
 
         (bytes memory payload, uint256 extraFee) = Assets.sendToken(
-            ASSET_HUB_PARA_ID, assetHubAgent, token, msg.sender, destinationChain, destinationAddress, amount
+            address(this),
```

Comment:
See my comments above. Libraries can get address of gateway via `address(this)`.

```suggestion
```

Reply:
addressed in https://github.com/Snowfork/snowbridge/pull/904/commits/8289362594510c9672b87bd44bfbf09dcbe3f596